test/test-mc-range-encoding.c

```diff
@@ -470,6 +474,25 @@ static void _test_RangeTest_Encode_Double(_mongocrypt_tester_t *tester) {
                            // Applying min/max/precision result in a domain needing >= 64 bits to represent.
                            // For range v2, expect an error.
                            .expectError = "Invalid upper bounds for double precision."},
+                          /* Test max and min integer bounds for doubles */
+                          {.value = DOUBLE_MIN_SAFE_INT,
+                           .max = OPT_DOUBLE_C(DOUBLE_MAX_SAFE_INT),
+                           .min = OPT_DOUBLE_C(DOUBLE_MIN_SAFE_INT),
+                           .precision = OPT_U32_C(0),
+                           .expect = 0,
+                           .expectMax = OPT_U64_C(36028797018963967)},
+                          {.value = DOUBLE_MIN_SAFE_INT + 1,
+                           .max = OPT_DOUBLE_C(DOUBLE_MAX_SAFE_INT),
+                           .min = OPT_DOUBLE_C(DOUBLE_MIN_SAFE_INT),
+                           .precision = OPT_U32_C(0),
+                           .expect = 1,
+                           .expectMax = OPT_U64_C(36028797018963967)},
+                         {.value = DOUBLE_MAX_SAFE_INT,
```

Comment:
Add test cases for value 0-10. Doesn't matter what the actual expect is (should be 9007199254740992-9007199254741002) but I want to know that they are all different values.